src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/service/JJDisputeServiceTest.java

```diff
@@ -0,0 +1,91 @@
+		jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+	}
+
+	@ParameterizedTest
+	@EnumSource(value = JJDisputeStatus.class, names = { "COMPLETED" })
+	void testSetStatusToIN_PROGRESS_405(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDisputeToUpdate = saveDispute(jjDisputeStatus);
+		JJDispute jjDisputeWithUpdatedStatus = saveDispute(JJDisputeStatus.IN_PROGRESS);
+		assertThrows(NotAllowedException.class, () -> {
+			jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+		});
+	}
+	
+	@ParameterizedTest
+	@EnumSource(value = JJDisputeStatus.class, names = { "NEW", "REVIEW", "IN_PROGRESS" })
+	void testSetStatusToREVIEW_200(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDisputeToUpdate = saveDispute(jjDisputeStatus);
+		JJDispute jjDisputeWithUpdatedStatus = saveDispute(JJDisputeStatus.REVIEW);
+		jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+	}
+
+	@ParameterizedTest
+	@EnumSource(value = JJDisputeStatus.class, names = { "COMPLETED" })
+	void testSetStatusToREVIEW_405(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDisputeToUpdate = saveDispute(jjDisputeStatus);
+		JJDispute jjDisputeWithUpdatedStatus = saveDispute(JJDisputeStatus.REVIEW);
+		assertThrows(NotAllowedException.class, () -> {
+			jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+		});
+	}
+	
+	@ParameterizedTest
+	@EnumSource(value = JJDisputeStatus.class, names = { "REVIEW" })
+	void testSetStatusToCOMPLETED_200(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDisputeToUpdate = saveDispute(jjDisputeStatus);
+		JJDispute jjDisputeWithUpdatedStatus = saveDispute(JJDisputeStatus.COMPLETED);
+		jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+	}
+
+	@ParameterizedTest
+	@EnumSource(value = JJDisputeStatus.class, names = { "NEW", "COMPLETED", "IN_PROGRESS" })
+	void testSetStatusToCOMPLETED_405(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDisputeToUpdate = saveDispute(jjDisputeStatus);
+		JJDispute jjDisputeWithUpdatedStatus = saveDispute(JJDisputeStatus.COMPLETED);
+		assertThrows(NotAllowedException.class, () -> {
+			jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+		});
+	}
+
+	@Test
+	void testSetStatusToNEW_405() {
+		JJDispute jjDisputeToUpdate = saveDispute(JJDisputeStatus.NEW);
+		JJDispute jjDisputeWithUpdatedStatus = saveDispute(JJDisputeStatus.NEW);
+		assertThrows(NotAllowedException.class, () -> {
+			jjDisputeService.updateJJDispute(jjDisputeToUpdate.getTicketNumber(), jjDisputeWithUpdatedStatus);
+		});
+	}
+
+	private JJDispute saveDispute(JJDisputeStatus jjDisputeStatus) {
+		JJDispute jjDispute = RandomUtil.createJJDispute();
+		jjDispute.setStatus(jjDisputeStatus);
+
+		return jjDisputeRepository.save(jjDispute);
+	}
+
+}
```

Comment:
Well done. Nice tests.

Reply:
Thanks, it was mostly inspired from your tests for dispute service :)